File: test/ayush.dev/scala/pipelines/scala_external_pipeline/code/src/main/scala/io/prophecy/pipelines/scala_external_git_pipeline/graph/Join_1.scala
package io.prophecy.pipelines.scala_external_git_pipeline.graph

import io.prophecy.libs._
import io.prophecy.pipelines.scala_external_git_pipeline.config.ConfigStore._
import io.prophecy.pipelines.scala_external_git_pipeline.udfs.UDFs._
import io.prophecy.pipelines.scala_external_git_pipeline.udfs._
import org.apache.spark._
import org.apache.spark.sql._
import org.apache.spark.sql.functions._
import org.apache.spark.sql.types._
import org.apache.spark.sql.expressions._
import java.time._

object Join_1 {

  def apply(spark: SparkSession, in0: DataFrame, in1: DataFrame): DataFrame =
    in0
      .as("in0")
      .join(in1.as("in1"), col("in0.a") === col("in1.a"), "inner")
      .select(col("in0.a").as("a"))

}

File: test/ayush.dev/python/pipelines/PythonInsideProjectDatasetsOnly/code/pythoninsideprojectdatasetsonly/graph/Join_1.py
from pyspark.sql import *
from pyspark.sql.functions import *
from pyspark.sql.types import *
from pythoninsideprojectdatasetsonly.config.ConfigStore import *
from pythoninsideprojectdatasetsonly.udfs.UDFs import *

def Join_1(spark: SparkSession, in0: DataFrame, in1: DataFrame, ) -> DataFrame:
    return in0\
        .alias("in0")\
        .join(in1.alias("in1"), (col("in0.c_tinyint") == col("in1.c_tinyint")), "inner")\
        .select(col("in0.c_tinyint").alias("c_tinyint"), col("in0.c_int").alias("c_int"), col("in0.c_boolean").alias("c_boolean"))
